src/coreclr/vm/excep.cpp

```diff
@@ -6767,10 +6767,12 @@ VEH_ACTION WINAPI CLRVectoredExceptionHandler(PEXCEPTION_POINTERS pExceptionInfo
 
     if (pExceptionInfo->ExceptionRecord->ExceptionCode == STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT)
     {
-        if (pThread == NULL || !pThread->PreemptiveGCDisabled())
+        if (pThread == NULL ||
+            !pThread->PreemptiveGCDisabled() ||
+            !pThread->HasThreadState(Thread::TS_Hijacked))
```

Comment:
What is the situation that the `TS_Hijacked` check is trying to handle?

Reply:
If we did not hijack the thread, we cannot continue here. We can't unhijack it. 
If we see such state, we can either crash or just let this be handled by whoever did the hijack. 

There is a theoretical possibility that someone else hijacked our thread, even if we were running in coop mode. Managed runtimes (dotnet or whatever) might not do that, but there is still a possibility. Maybe something like debugger or profiler could do this. 

This style of hijacking does not allow to tell who really hijacked a thread. On the other hand if a thread was hijacked twice, by different entities, I think things could work.

Reply:
Two entities trying to hijack the same return address would be a factory for race conditions. If we detect that somebody else tried to hijack our return address, I think the best action is immediate fail fast.

Reply:
fail fast would work too. 

One thing is certain - if the hijacked thread (or more precisely - the hijacked location) was not hijacked by us, we cannot continue on this path.